src/score_hv/harvesters/gsi_satellite_radiance.py

```diff
@@ -0,0 +1,415 @@
+                if len(series_num_list & channel_stats_dict.keys()) > 0:
+                    """this obs_type has active channels
+                    return stats by channel for this obs_type
+                    """
+                    for stat in self.config.stats_to_harvest:
+                        values_by_channel = list()
+                        longnames = list()
+                        for channel_index, series_number in enumerate(
+                                                               series_num_list):
+                            if series_number in channel_stats_dict.keys():
+                                """active channel!
+                                """
+                                channel_stat_dict = channel_stats_dict[
+                                                            series_number][stat]
+                                
+                                assert obs_type == channel_stat_dict[
+                                                             'observation_type']
+                                assert self.obs_type_channels[
+                                    obs_type][channel_index
+                                    ] == channel_stat_dict['channel_number']
+                                    
+                                values_by_channel.append(
+                                                    channel_stat_dict['value'])
+                                longnames.append(channel_stat_dict['longname'])
+                            else:
+                                """active obs_type but inactive channel
+                                """
+                                values_by_channel.append(None)
+                                longnames.append(None)
+                        
+                        self.satinfo_stats.append(SatinfoStat(
+                            self.datetime,
+                            gsi_stage,
+                            obs_type,
+                            series_num_list,
+                            self.obs_type_channels[obs_type],
+                            stat,
+                            values_by_channel,
+                            longnames
+                        ))
+        
+        return self.satinfo_stats
+
+    def get_channel_stats(self, line_parts):
+        """iterate through the requested statistics and extract relevant data
+        """
+        series_number = int(line_parts[0]) # from satinfo file, index from 1 
+        #channel_index = series_number - 1 # from self.channels, index from 0
+        channel_number = int(line_parts[1])
+        observation_type = line_parts[2]
+        
+        # make sure we have the correct channel
+        #assert self.channels[channel_index].series_number==series_number
+        assert self.channels[series_number]['observation_type']==observation_type
+        assert self.channels[series_number]['channel'] == channel_number      
```

Comment:
same concern about the asserts above, is this something that should completely crash the code and should there be a custom exception to clarify what the failure here is?

Reply:
replaced with if block -> raise ValueError if false